### PR TITLE
SI-7421 remove unneeded extra-attachement in maven deploy

### PR DESCRIPTION
**Pull request by huitseeker**

### CODE REVIEW

**Comment by huitseeker** [comment]
Review by @dragos.

File: src/build/maven/maven-deploy.xml [hunk 1]
```diff
@@ -164,11 +164,7 @@
       <attribute name="repository" />
       <attribute name="version" />
       <sequential>
-        <deploy-remote name="scala-library" version="@{version}" repository="@{repository}">
-          <extra-attachments>
-            <artifact:attach type="jar" file="scala-library/scala-library-docs.jar" classifier="javadoc" />
-          </extra-attachments>
-        </deploy-remote>
+        <deploy-remote name="scala-library" version="@{version}" repository="@{repository}"/>
```
Comment (jsuereth) [inline src/build/maven/maven-deploy.xml:167]: deploy-remote includes src docs by default now?  I think this went stale when "deploy-remote-signed" showed up...

**Comment by jsuereth** [comment]
I think this LGTM.

**Comment by dragos** [comment]
LGTM too

**Comment by paulp** [comment]
That's a lot of LGTYness.

**Pull request merged**